CXX-3502 Fix explicit "upsert: false" behavior for "findOneAnd*" commands (#1652)
mongodb/mongo-cxx-driver: CHANGELOG.md

### Fixed

> - Correctly set `upsert: false` for "findOneAnd*" commands when the option is explicitly set to `false` (regression in 4.2.0).

Reviewer: ```suggestion
- Do not set `upsert` to `true` in `find_one_and_*` operations when the option is explicitly set to `false` (regression in 4.2.0).
```

Suggest tweaking message to match expected behavior.

The C driver only applies `upsert: true` to the `findAndModify` command if the flag [is set](https://github.com/mongodb/mongo-c-driver/blob/66cf09c0f26ad13885daf331e419d1df45857275/src/libmongoc/src/mongoc/mongoc-collection.c#L2315-L2317). I expect this bug caused `upsert: true` to be sent if the the C++ driver option was explicitly set to false.

I also associate "command" with the server command (`findAndModify`) and "operation" with the driver API (`find_one_and_*`).
